# HG changeset patch
# User Andrew Cooper <andrew.cooper3@citrix.com>
# Date 1394717782 -3600
# Node ID 40b60c5823ea9eb5c0d4c2400f650139ad7c34d9
# Parent  d240dab2fe5b2e024a16b4284188fa151fee517e
time: move wallclock_time() declaration into common code

It is called from common code, but has architecture specific implementations.
Have one declaration instead of two.

No functional change.

Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>
Acked-by: Ian Campbell <ian.campbell@citrix.com>
Acked-by: Keir Fraser <keir@xen.org>

diff -r d240dab2fe5b -r 40b60c5823ea xen/include/asm-arm/time.h
--- a/xen/include/asm-arm/time.h
+++ b/xen/include/asm-arm/time.h
@@ -12,9 +12,6 @@ static inline cycles_t get_cycles (void)
         return 0;
 }
 
-struct tm;
-struct tm wallclock_time(void);
-
 /* List of timer's IRQ */
 enum timer_ppi
 {
diff -r d240dab2fe5b -r 40b60c5823ea xen/include/asm-x86/time.h
--- a/xen/include/asm-x86/time.h
+++ b/xen/include/asm-x86/time.h
@@ -48,9 +48,6 @@ int dom0_pit_access(struct ioreq *ioreq)
 
 int cpu_frequency_change(u64 freq);
 
-struct tm;
-struct tm wallclock_time(void);
-
 void pit_broadcast_enter(void);
 void pit_broadcast_exit(void);
 int pit_broadcast_is_available(void);
diff -r d240dab2fe5b -r 40b60c5823ea xen/include/xen/time.h
--- a/xen/include/xen/time.h
+++ b/xen/include/xen/time.h
@@ -48,6 +48,7 @@ struct tm {
     int     tm_isdst;       /* daylight saving time */
 };
 struct tm gmtime(unsigned long t);
+struct tm wallclock_time(void);
 
 #define SYSTEM_TIME_HZ  1000000000ULL
 #define NOW()           ((s_time_t)get_s_time())
